lib/server.go

```diff
@@ -862,7 +956,17 @@ func (srv *Server) GetBlocksToStore(pp *Peer) {
 	}
 
 	// Go through the block nodes in the blockchain and download the blocks if they're not stored.
-	for _, blockNode := range srv.blockchain.bestChain {
+	for ii := uint32(srv.blockchain.lowestBlockNotStored); ii <= srv.blockchain.blockTip().Height; ii++ {
```

Comment:
@diamondhands0 - we'll probably want to rewrite this so that we iterate in reverse. This allows us to more quickly look up blocks since we have both the block hash and block height.

Reply:
Yeah, that makes sense. I'd only do it if it's very slow right now, though. If it's bearable (eg only adds 20m or something like that to an initial mainnet sync) then I think it's fine to leave it.